app/controllers/components/course/lesson_plan_component.rb

```diff
@@ -20,7 +20,7 @@ def main_sidebar_items
     [
       {
         key: :lesson_plan,
-        icon: 'book',
+        icon: :lessonPlan,
```

Comment:
I am. `icon`s are camelCase because the existing `COURSE_COMPONENT_ICONS` object's key is in camelCase, and this value is mapped directly to that object.

The `icon`s are (mostly) singular because the existing `COURSE_COMPONENT_ICONS` object's key were in singular too, e.g., `assessment` and `achievement`. As such, I followed by singularising everything else, e.g., `announcement`, `submission`, etc., except for a few that are clearly plural, e.g., `settings`, `levels`, `groups`, `comments`, `manageUsers`, etc.

The `key`s are snake_case because the existing keys are all in snake_case. I only added a few (3?) missing ones. I think it's fine if it's snake_case, too, since it's not used code-wise in the front-end, other than as React keys.